src/main/kotlin/JobState.kt

```diff
@@ -0,0 +1,64 @@
+package franz.internal
+
+fun <T, U> JobDSL<T, U>.asPipe(): JobState<U> = JobState(this.value)
```

Comment:
If we refer to the concept as a pipe in JobDSL then our class should be called as pipe.

2. If we want to keep JobDSL.Kt clean from dependencies on pipe then the pipe stuff should have its own namespace.